src/rapid_pro_to_engagement_db/rapid_pro_to_engagement_db.py

```diff
@@ -174,15 +172,20 @@ def sync_rapid_pro_to_engagement_db(rapid_pro, engagement_db, uuid_table, rapid_
     # (If the cache or a contacts file for this workspace don't exist, `contacts` will be `None` for now)
     contacts = _get_contacts_from_cache(cache)
 
-    dataset_to_sync_stats = dict()  # of '{flow_name}.{flow_result_field}' -> RapidProToEngagementDBSyncStats
-    for flow_config in rapid_pro_config.flow_result_configurations:
-        sync_stats = RapidProToEngagementDBSyncStats()
+    groups = []
+    unique_keys = []
+    flow_result_configurations = sorted(rapid_pro_config.flow_result_configurations, key=lambda x: x.flow_name)
+    for flow_name, flow_configs in groupby(flow_result_configurations, lambda x: x.flow_name):
+        groups.append(list(flow_configs))
+        unique_keys.append(flow_name)
+
+    flow_stats = dict() # of flow_name -> FlowStats
+    dataset_to_sync_stats = defaultdict(lambda: FlowResultToEngagementDBSyncStats())  # of '{flow_name}.{flow_result_field}' -> FlowResultToEngagementDBSyncStats
+    for flow_name, flow_configs in zip(unique_keys, groups):
```

Comment:
Instead of creating 2 lists then zipping them together, I think you could use a dictionary instead. That way there's no risk of the lists ever getting out of sync with each other before you zip

a rough sketch of this would look like:
```python
flow_name_to_flow_configs = dict()
for flow_result_config in flow_result_configurations:
    flow_name_to_flow_configs[flow_result_config.flow_name] = flow_result_config

for flow_name, flow_configs in flow_name_to_flow_configs.items():
    ...
```